appmonitoring/ts/src/CertificateManager.ts

```diff
@@ -155,22 +155,38 @@ export class CertificateManager {
         }
     }
 
-    public async GetSecretDetails(operationId: string, kubeConfig: k8s.KubeConfig): Promise<WebhookCertData> {
+    public async GetSecretDetails(operationId: string, kubeConfig: k8s.KubeConfig, clusterArmId: string, clusterArmRegion: string): Promise<WebhookCertData> {
         try {
             const k8sApi = kubeConfig.makeApiClient(k8s.CoreV1Api);
             const secretsObj: k8s.V1Secret = await k8sApi.readNamespacedSecret({ name: CertificateStoreName, namespace: KubeSystemNamespaceName });
+            let certificate: WebhookCertData = null;
             if (secretsObj.data) {
-                const certificate: WebhookCertData = {
-                    caCert: Buffer.from(secretsObj.data['ca.cert'], 'base64').toString('utf-8'),
-                    caKey: Buffer.from(secretsObj.data['ca.key'], 'base64').toString('utf-8'),
-                    tlsCert: Buffer.from(secretsObj.data['tls.cert'], 'base64').toString('utf-8'),
-                    tlsKey: Buffer.from(secretsObj.data['tls.key'], 'base64').toString('utf-8')
-                };
+                try {
+                    certificate = {
+                        caCert: Buffer.from(secretsObj.data['ca.cert'], 'base64').toString('utf-8'),
+                        caKey: Buffer.from(secretsObj.data['ca.key'], 'base64').toString('utf-8'),
+                        tlsCert: Buffer.from(secretsObj.data['tls.cert'], 'base64').toString('utf-8'),
+                        tlsKey: Buffer.from(secretsObj.data['tls.key'], 'base64').toString('utf-8')
+                    };
+                } catch (error) {
+                    logger.error('Failed to decode secret data from base64!', operationId, this.requestMetadata);
+                    logger.error(JSON.stringify(error), operationId, this.requestMetadata);
+                    await logger.SendEvent("CertificateBase64DecodeFailure", operationId, null, clusterArmId, clusterArmRegion, true, error);
+                    
+                    certificate = {
+                        caCert: null,
+                        caKey: null,
+                        tlsCert: null,
+                        tlsKey: null
+                    }
+                    logger.info('Returning empty certificate data due to decode failure.', operationId, this.requestMetadata);
+                }
 
                 return certificate;
             }
         } catch (error) {
             logger.error(JSON.stringify(error), operationId, this.requestMetadata);
```

Comment:
Tell us what failed exactly. Which call? readNamespacedSecret?

Reply:
Yes, but also tell us the context of the failure:
`Error reading the secret: ${JSON.stringify(error)}`